Geometry/MTDCommonData/test/TestMTDPosition.cc

```diff
@@ -0,0 +1,206 @@
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+
+#include "DetectorDescription/Core/interface/DDValue.h"
+#include "DetectorDescription/Core/interface/DDCompactView.h"
+#include "DetectorDescription/Core/interface/DDExpandedNode.h"
+#include "DetectorDescription/Core/interface/DDExpandedView.h"
+#include "DetectorDescription/Core/interface/DDLogicalPart.h"
+#include "DetectorDescription/Core/interface/DDSolid.h"
+#include "DetectorDescription/Core/interface/DDSolidShapes.h"
+
+//#define EDM_ML_DEBUG
+
+class TestMTDPosition : public edm::one::EDAnalyzer<> {
+public:
+  explicit TestMTDPosition( const edm::ParameterSet& );
+  ~TestMTDPosition() override;
+
+  void beginJob() override {}
+  void analyze(edm::Event const&, edm::EventSetup const&) override;
+  void endJob() override {}
+
+  void theBaseNumber( const DDGeoHistory& gh );
+
+  void checkMTD( const DDCompactView& cpv, std::string fname = "GeoHistory", int nVols = 0 , std::string ddtop_ = "mtd:BarrelTimingLayer" );
+
+private:
+
+  std::string label_;
+  bool isMagField_;
+  std::string fname_;
+  int nNodes_;
+  std::string ddTopNodeName_;
+
+  static constexpr double rad2deg = 180./M_PI;
+
+};
+
+TestMTDPosition::TestMTDPosition( const edm::ParameterSet& iConfig ) :
+  label_(iConfig.getUntrackedParameter<std::string>("label","")),
+  isMagField_(iConfig.getUntrackedParameter<bool>("isMagField",false)),
+  fname_(iConfig.getUntrackedParameter<std::string>("outFileName", "GeoHistory")),
+  nNodes_(iConfig.getUntrackedParameter<uint32_t>("numNodesToDump", 0)),
+  ddTopNodeName_(iConfig.getUntrackedParameter<std::string>("ddTopNodeName", "btl:BarrelTimingLayer"))
+{
+  if ( isMagField_ ) {
+    label_ = "magfield";
+  }
+}
+
+TestMTDPosition::~TestMTDPosition()
+{
+}
+
+void
+TestMTDPosition::analyze( const edm::Event& iEvent, const edm::EventSetup& iSetup )
+{
+   using namespace edm;
+
+   edm::ESTransientHandle<DDCompactView> pDD;
+   if (!isMagField_) {
+     iSetup.get<IdealGeometryRecord>().get(label_, pDD );
+   } else {
+     iSetup.get<IdealMagneticFieldRecord>().get(label_, pDD );
+   }
+   if (pDD.description()) {
+     edm::LogInfo("TestMTDPosition") << pDD.description()->type_ << " label: " << pDD.description()->label_;
+   } else {
+     edm::LogWarning("TestMTDPosition") << "NO label found pDD.description() returned false.";
+   }
+   if (!pDD.isValid()) {
+     edm::LogError("TestMTDPosition") << "ESTransientHandle<DDCompactView> pDD is not valid!";
+   }
+
+   if ( ddTopNodeName_ != "btl:BarrelTimingLayer" && ddTopNodeName_ != "etl:EndcapTimingLayer" ) {
+     edm::LogWarning("TestMTDPosition") << ddTopNodeName_ << "Not valid top MTD volume";
+     return;
+   }
+     
+   checkMTD( *pDD , fname_ , nNodes_ , ddTopNodeName_ );
+
+}
+
+void TestMTDPosition::checkMTD ( const DDCompactView& cpv, std::string fname, int nVols , std::string ddtop_ ) {
+
+  fname = "dump" + fname;
+  DDExpandedView epv(cpv);
```

Comment:
@fabiocos - I think, you want to use DDFilteredView here

Reply:
@ianna I based myself on https://github.com/cms-sw/cmssw/blob/master/DetectorDescription/OfflineDBLoader/src/GeometryInfoDump.cc#L29
as for the other TestMTDNumbering class...

Reply:
@ianna do you have a specific suggestion about the filter, some example to which point me?

Reply:
@fabiocos - here is an analyzer example:
https://cmssdt.cern.ch/lxr/source/Geometry/CMSCommonData/test/DDFilteredViewAnalyzer.cc#0050
which you can run as:
```
process.fva = cms.EDAnalyzer("DDFilteredViewAnalyzer",
                             attribute = cms.string("OnlyForHcalSimNumbering"),
                             value = cms.string("any"))
```
It would be better to pass:
```
using Position = Surface::PositionType;
using Rotation = Surface::RotationType;
```